packages/test/test-end-to-end-tests/src/test/SummarizeFromLatest.spec.ts

```diff
@@ -0,0 +1,311 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "assert";
+import {
+    DataObject,
+    DataObjectFactory,
+    PureDataObject,
+} from "@fluidframework/aqueduct";
+import { IContainer } from "@fluidframework/container-definitions";
+import {
+    IContainerRuntimeOptions,
+    ISummarizer,
+} from "@fluidframework/container-runtime";
+import { FluidObject, IFluidHandle, IRequest } from "@fluidframework/core-interfaces";
+import { FluidDataStoreRuntime, mixinSummaryHandler } from "@fluidframework/datastore";
+import { SharedMatrix } from "@fluidframework/matrix";
+import { SharedMap } from "@fluidframework/map";
+import { requestFluidObject } from "@fluidframework/runtime-utils";
+import { ITestObjectProvider,
+    waitForContainerConnection,
+    summarizeNow,
+    createSummarizerFromFactory,
+} from "@fluidframework/test-utils";
+import { describeNoCompat, getContainerRuntimeApi } from "@fluidframework/test-version-utils";
+import { IContainerRuntimeBase, IFluidDataStoreFactory } from "@fluidframework/runtime-definitions";
+import { pkgVersion } from "../packageVersion";
+
+interface ProvideSearchContent {
+    SearchContent: SearchContent;
+  }
+interface SearchContent extends ProvideSearchContent {
+    getSearchContent(): Promise<string | undefined>;
+  }
+
+// Note GC needs to be disabled.
+const runtimeOptions: IContainerRuntimeOptions = {
+    summaryOptions: {
+        disableSummaries: true,
+        summaryConfigOverrides: { state: "disabled" },
+    },
+    gcOptions: { gcAllowed: false },
+};
+export const TestDataObjectType1 = "@fluid-example/test-dataStore1";
+export const TestDataObjectType2 = "@fluid-example/test-dataStore2";
+class TestDataObject2 extends DataObject {
+    public get _root() {
+        return this.root;
+    }
+    public get _context() {
+        return this.context;
+    }
+    private readonly mapKey = "SharedMap";
+    public map!: SharedMap;
+
+    protected async initializingFirstTime() {
+        const sharedMap = SharedMap.create(this.runtime, this.mapKey);
+        this.root.set(this.mapKey, sharedMap.handle);
+   }
+
+    protected async hasInitialized() {
+        const mapHandle = this.root.get<IFluidHandle<SharedMap>>(this.mapKey);
+        assert(mapHandle !== undefined, "SharedMap not found");
+        this.map = await mapHandle.get();
+    }
+}
+
+class TestDataObject1 extends DataObject implements SearchContent {
+    public async getSearchContent(): Promise<string | undefined> {
```

Comment:
Testing loading other data stores is important, but should it be part of that flow (that tests acks and reloads)? Feels like it should be tested separately

Reply:
we do not need it for this particular test -  I cleaned it up.